src/pybind/ceph_volume_client.py

```diff
@@ -582,8 +583,11 @@ def authorize(self, volume_path, auth_id):
         # Now construct auth capabilities that give the guest just enough
         # permissions to access the share
         client_entity = "client.{0}".format(auth_id)
-        want_mds_cap = 'allow rw path={0}'.format(path)
-        want_osd_cap = 'allow rw pool={0} namespace={1}'.format(pool_name, namespace)
+        access_level = ('r' if readonly else 'rw')
+        want_mds_cap = 'allow {0} path={1}'.format(access_level, path)
+        want_osd_cap = 'allow {0} pool={1} namespace={2}'.format(access_level,
+                                                                 pool_name,
+                                                                 namespace)
```

Comment:
Actually, come to think of it, I think this already behaves the way @xiaoxichen suggests, on the authorize() path.  If someone calls this twice for the same ID, once with readonly=True and once with readonly=False, the result will be two rules.  At the ceph-mds level, these will be interpreted as whichever is broader winning (i.e. if both are present, then it will be RW)